source/i2c_api.c

```diff
@@ -321,4 +322,185 @@ void i2c_reset(i2c_t *obj)
     }
 }
 
+#if DEVICE_I2CSLAVE
```

Comment:
Is this a good enough reason to remove this part of the code?!? In my opinion it isn't.